File: src/test/scala/tests/graphics/TestSprite.scala
package tests

import sfml.graphics.*
import sfml.window.*

object TestSprite extends SNTest:
    override def snTest(snTestScreen: TestScreen): Unit =
        scala.util.Using.Manager { use =>
            // Setup
            val window = use(RenderWindow(VideoMode(1024, 768), "Test"))

            val texture = use(Texture())
            texture.loadFromFile("src/test/resources/sfml.png")

            val sprite = use(Sprite(texture))

            window.isOpen()

            // Control test
            window.clear()
            window.draw(sprite)
            window.display()
            snTestScreen.takeScreenshot()

            // sf::Sprite::setPosition
            sprite.position = (100, 100)

            window.clear()
            window.draw(sprite)
            window.display()
            snTestScreen.takeScreenshot()

            // Teardown
            window.closeWindow()
        }

File: src/test/cxx/tests/graphics/TestSprite.cpp
#include "Test.hpp"

#include <SFML/Graphics.hpp>

snTest(TestSprite)
{
    // Setup
    sf::RenderWindow window(sf::VideoMode(1024, 768), "Test");

    sf::Texture texture;
    texture.loadFromFile("src/test/resources/sfml.png");

    sf::Sprite sprite(texture);

    window.isOpen();

    // Control test
    window.clear();
    window.draw(sprite);
    window.display();
    snTestScreen.takeScreenshot();

    // sf::Sprite::setPosition
    sprite.setPosition(100, 100);

    window.clear();
    window.draw(sprite);
    window.display();
    snTestScreen.takeScreenshot();

    // Teardown
    window.close();
}
